spec/inputs/beats/message_listener_spec.rb

```diff
@@ -52,10 +52,74 @@ def flush(&block)
   end
 end
 
+shared_examples "when the message is from any libbeat" do |ecs_compatibility, host_field_name|
+  let(:input) do
+    input = LogStash::Inputs::Beats.new({ "port" => 5555, "codec" => codec, "ecs_compatibility" => "#{ecs_compatibility}" })
+    input.register
+    input
+  end
+
+  #Requires data modeled as Java, not Ruby since the actual code pulls from Java backed (Netty) object
+  let(:data) do
+    d = HashMap.new
+    d.put('@metadata', HashMap.new)
+    d.put('metric',  1)
+    d.put('name', "super-stats")
+    d
+  end
+
+  let(:message) { MockMessage.new("abc",  data)}
+
+  it "extract the event" do
+    subject.onNewMessage(ctx, message)
+    event = queue.pop
+    expect(event.get("message")).to be_nil
+    expect(event.get("metric")).to eq(1)
+    expect(event.get("name")).to eq("super-stats")
+    puts "DNADBG>> ecs_compatibility: #{ecs_compatibility} host_field_name: #{host_field_name}, event: #{event.to_hash}"
```

Comment:
:scissors:

Reply:
I do feel your pain, think we might have "over-engineered" the specs with the ecs_compatibility meta stuff.
let's see how it plays out in the long run, personally rather have duplicity in specs than having to :thought_balloon: on how it works...